code/game/objects/structures/tables_racks.dm

```diff
@@ -720,24 +720,16 @@
 	buckle_lying = 90
 	climbable = FALSE
 	custom_materials = list(/datum/material/silver =SHEET_MATERIAL_AMOUNT)
-	var/mob/living/carbon/patient = null
-	var/obj/machinery/computer/operating/computer = null
 
 /obj/structure/table/optable/Initialize(mapload)
 	. = ..()
-	for(var/direction in GLOB.alldirs)
-		computer = locate(/obj/machinery/computer/operating) in get_step(src, direction)
-		if(computer)
-			computer.table = src
-			break
 
-	RegisterSignal(loc, COMSIG_ATOM_ENTERED, PROC_REF(mark_patient))
-	RegisterSignal(loc, COMSIG_ATOM_EXITED, PROC_REF(unmark_patient))
+	AddComponent(/datum/component/links_to_operating_computers, provide_upgraded_surgeries = TRUE)
+
+	RegisterSignal(loc, COMSIG_ATOM_ENTERED, PROC_REF(on_atom_entered))
+	RegisterSignal(loc, COMSIG_ATOM_EXITED, PROC_REF(on_atom_exited))
```

Comment:
Shouldn't this be using connect loc

Reply:
I decided not to on the belief that if an operating table is inside some theoretical "thing", then it shouldn't connect to anything in the outside world

Reply:
Oh right I forgot connect_loc doesn't go to the turf...either way we are doing more work than just registering signals so it would be easier to keep it like this on that fact rather than also try to have connect_loc in the mix

Reply:
Still seems like it creates obvious problems like leaving behind the signals on the old loc if moved (admin throw mode / admin get, for example)

Reply:
Didn't realize you were referring to the existing opreating table code, fixing